### PR TITLE
Bug directory creation

### PR DESCRIPTION
**Pull request by deanott**
Fixes with more catches that actually returned the promise rejectsions. 

The mkdir function is also now correctly using the old fix. 

The downloading process can still return errors with process but there due to the token generation.

There may be more handling to be done but refactoring to remove the downloader call from being there anyway.

### CODE REVIEW

**Comment by deanott** [comment]
@AdamSlack you can accept this also, so do so...

File: pipeline/scraper/scrape.js [hunk 1]
```diff
@@ -28,17 +28,22 @@ const logger = require('./logger.js');
 
 let appsSaveDir = path.join(config.datadir, 'apps');
 
-function mkdirp(path) {
-    path.split(path.sep).reduce((parentDir, childDir) => {
+
+function catchErr(msg, err, func) {
+    func(msg,err);
+    return Promise.reject();
+}
```
Comment (sauyon) [inline pipeline/scraper/scrape.js]: Add a newline here

File: pipeline/scraper/scrape.js [hunk 2]
```diff
@@ -28,17 +28,22 @@ const logger = require('./logger.js');
 
 let appsSaveDir = path.join(config.datadir, 'apps');
 
-function mkdirp(path) {
-    path.split(path.sep).reduce((parentDir, childDir) => {
+
+function catchErr(msg, err, func) {
```
Comment (sauyon) [inline pipeline/scraper/scrape.js]:
```
function catchErr(msg, func) {
    return function(err) {
        func(msg, err);
        return Promise.reject();
    }
}

File: pipeline/scraper/scrape.js [hunk 3]
```diff
@@ -130,7 +137,7 @@ function extractAppData(appData) {
                 try {
                     fs.rmdir(appSavePath);
                 } catch (err) {
-                    // TODO: something
+                    return catchErr('Could not remove app saved path', err, logger.err);
```
Comment (sauyon) [inline pipeline/scraper/scrape.js]: Don't return a promise here.

**Review by sauyon — COMMENTED** [review]
Just a few things

**Review by sauyon — APPROVED** [review]
Approved.

**Pull request merged**